staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -552,6 +552,8 @@ type VirtualMachineInstanceNetworkInterface struct {
 	InterfaceName string `json:"interfaceName,omitempty"`
 	// Specifies the origin of the interface data collected. values: domain, guest-agent, or both
 	InfoSource string `json:"infoSource,omitempty"`
+	// Specifies how many queues are allocated by MultiQueue, 0 if can't be determined
+	QueueCount uint `json:"queueCount"`
```

Comment:
IMO a good API will try to make it easy for an API consumer to work with it without asking questions about existence.
All network interfaces have a queue, be it 1 or more. We can in this case always report it and make it simple to consume.

Is there a disadvantage to this approach?
The only corner case I can think of (but unsure if it can manifest itself) is that an older virt-handler will try to update the status of a newer VMI.

Reply:
if we don't have accurate information, we shouldn't be advertising any information. this is why we have optional structs and  fields in the status. i don't think queues are a special case.

Reply:
We have accurate information from the domain. We may not have information if the interface reported by the guest-agent cannot be associated with an interface defined in the domain, which is a corner case.

I argue that this should not be an optional field in a report. The user should not need to know what is the default if not specified (0 or 1).
I will not fight over adding here `omitempty`, but I think it is wrong if it is not a required by something else.
As an API consumer the missing key are IMO a heavy burden on its consumption.

@AlonaKaplan , WDYT?

Reply:
IIUC, `QueueCount` will be 0 only if we couldn't get the real number of the queues. In this case I find it more user friendly to use `omitempty` and not show it at all. I think showing `queueCount 0` is confusing. What should the user understand when see there are `0` queues? Not showing the value reflects better the fact we don't know what is the value.

Reply:
You understood correct
Switching to omitEmpty then

Reply:
excellent, yes it was the `zero` case i was most concerned about, which i agree that omitempty helps with in a way that is consistent with the rest of our api.